python/rbac/services/role.py

```diff
@@ -11,13 +11,35 @@
 # limitations under the License.
 
 from typing import List
-
+from adwp_base.errors import raise_AdwpEx as err
 from rest_framework.exceptions import ValidationError
 
 from rbac.models import Role, RoleTypes
 from rbac.utils import update_m2m_field
 
 
+def set_parametrized_from_child(role, children: List[Role]):
+    param_list = []
+
+    for child in children:
+        child_params = child.parametrized_by_type
+        for child_param in child_params:
```

Comment:
It seems a bit overcomplicated. It could be smth like that:
paramset = set()
for ch in children:
    paramset.update(ch.parametrize_by_type)
if paramset.intersection(provider_hierarchy) and paramset.intersection(cluster_hierarchy):
    raise AdwpEx(...)  # not err - look at its definition

Reply:
Changed